juspay/hyperswitch
# feat(process_tracker): make long standing payments failed

## Type of Change



## Description

This feature will make the payments that have been stuck in `Pending` attempt status because of connection timed out or some other 5xx error. These payments will not have the connector transaction id. So after the last retry by process tracker, these will be moved to failed.




## Motivation and Context



## How did you test it?

- Run producer and consumer locally.
- Simulate a connection timed out payment. Provide the connector url as https://google.com:81
- Wait for process tracker to make the status as `Failed` during the last retry.
<img width="1328" alt="Screenshot 2023-10-03 at 7 10 34 PM" src="https://github.com/juspay/hyperswitch/assets/48803246/ba304eb9-a9d0-4de7-9177-dafe7010dbaa">
<img width="1728" alt="Screenshot 2023-10-03 at 7 11 22 PM" src="https://github.com/juspay/hyperswitch/assets/48803246/9419b330-3613-48ff-8c2d-343f0ad8e1d9">



- Check if a failed webhook is received.

![Screenshot 2023-10-03 at 6 20 19 PM](https://github.com/juspay/hyperswitch/assets/48803246/5f493123-995e-4176-bcc2-39d0809f4400)


## Checklist
